StackExchange.Profiling/Helpers/ExtensionMethods.cs

```diff
@@ -10,6 +13,45 @@ namespace StackExchange.Profiling.Helpers
     public static class ExtensionMethods
     {
         /// <summary>
+        /// Performs the specified action on each element of the <see cref="T:System.Collections.Generic.IList`1"/>.
+        /// </summary>
+        /// <param name="list">the list to perform the operations for</param>
+        /// <param name="action">The <see cref="T:System.Action`1"/> delegate to perform on each element of the
+        ///  <see cref="T:System.Collections.Generic.List`1"/>.</param><exception cref="T:System.ArgumentNullException">
+        /// <paramref name="action"/> is null.</exception><exception cref="T:System.InvalidOperationException">An element in the collection has been modified. CautionThis exception is thrown starting with the .NET Framework 4.5. </exception>
+        public static void ForEach<T>(this IList<T> list, Action<T> action)
```

Comment:
i'm not very happy that this method is public, i would rather prefer if it was internal. and an alternative to this method is to update the couple of places which try to modify the list so they don't need ForEach and RemoveAll at all